cmd/juju/resource/deploy.go

```diff
@@ -121,19 +136,39 @@ func (d deployUploader) upload(files map[string]string, revisions map[string]int
 	return pending, nil
 }
 
-func (d deployUploader) checkFiles(files map[string]string) error {
-	for name, path := range files {
-		err := d.osStat(path)
-		if os.IsNotExist(err) {
-			return errors.Annotatef(err, "file for resource %q", name)
+func (d deployUploader) validateResourceDetails(resources map[string]string) error {
+	for name, value := range resources {
+		var err error
+		switch d.resources[name].Type {
+		case charmresource.TypeFile:
+			err = d.checkFile(name, value)
+		case charmresource.TypeDocker:
+			err = checkDockerDetails(name, value)
```

Comment:
The file check about actually checks that the file exists, as opposed to a simple syntax check. The docker equivalent should probably be to poke the URL to check for 404 etc. Ok to add a TODO for now but will need to be fixed.